### PR TITLE
feat: Add BSO Table

### PR DESCRIPTION
**Pull request by CIJones11**
## Description

This pull request is the creation of the BSO_ORGANISATIONS table, including its schema definition and corresponding column descriptions.

## Context

This change is required to define the BSO_ORGANISATIONS table, which will store organization-specific details, such as organization codes, recall intervals, age ranges, contact information, and process-related flags.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by MWClayson-NHS — DISMISSED** [review]
There are data types here that don't exist in SQL Server, Please make sure these changes work before raising a PR.

File: application/CohortManager/Set-up/database/create_statement.sql [hunk 1]
```diff
@@ -349,3 +349,95 @@ CREATE TABLE CURRENT_POSTING_LKP
...
+      BSO_REGION_ID INT,
+      ADMIN_EMAIL_ADDRESS VARCHAR(100),
+      IEP_DETAILS NVARCHAR(MAX),
+      NOTES NVARCHAR(MAX),
+      RLP_DATE_ENABLED DATETIME
+    );
+END
+
+-- Add description to the table
+EXEC sp_addextendedproperty
```
Comment (MWClayson-NHS) [inline application/CohortManager/Set-up/database/create_statement.sql:416]: Is there a reason we are adding extended property?

Comment (stephhou) [reply application/CohortManager/Set-up/database/create_statement.sql:416]: You can just delete the comments here Chris.

Comment (CIJones11) [reply application/CohortManager/Set-up/database/create_statement.sql]: Removed!

File: application/CohortManager/Set-up/database/create_statement.sql [hunk 2]
```diff
@@ -368,3 +368,64 @@ CREATE TABLE LANGUAGE_CODES
...
+      IS_AGEX_ACTIVE BIT NOT NULL,
+      AUTO_BATCH_LAST_RUN DATETIME,
+      AUTO_BATCH_MAX_DATE_TIME_PROCESSED DATETIME,
+      BSO_REGION_ID INT,
+      ADMIN_EMAIL_ADDRESS VARCHAR(100),
+      IEP_DETAILS NVARCHAR(MAX),
+      NOTES NVARCHAR(MAX),
+      RLP_DATE_ENABLED DATETIME
+    );
+END
```
Comment (andrewmorris43) [inline application/CohortManager/Set-up/database/create_statement.sql:431]: Needs a final newline on this file.

**Review by andrewmorris43 — COMMENTED** [review]
The table is created ok when I run the create statement, just needs the file format sorting and good to go.

File: application/CohortManager/Set-up/database/create_statement.sql [hunk 3]
```diff
@@ -368,3 +368,64 @@ CREATE TABLE LANGUAGE_CODES
...
+      FAILSAFE_MAX_AGE_YEARS DECIMAL(2) NOT NULL,
+      FAILSAFE_MAX_AGE_MONTHS DECIMAL(2) NOT NULL,
+      FAILSAFE_LAST_RUN DATETIME NOT NULL,
+      IS_AGEX BIT NOT NULL,
+      IS_AGEX_ACTIVE BIT NOT NULL,
+      AUTO_BATCH_LAST_RUN DATETIME,
+      AUTO_BATCH_MAX_DATE_TIME_PROCESSED DATETIME,
+      BSO_REGION_ID INT,
+      ADMIN_EMAIL_ADDRESS VARCHAR(100),
+      IEP_DETAILS NVARCHAR(MAX),
```
Comment (Warren-Pitterson) [inline application/CohortManager/Set-up/database/create_statement.sql:427]: this is a nVar char but could probably just be a varchar to keep inline with the rest of the tables, same with notes

File: application/CohortManager/Set-up/database/create_statement.sql [hunk 4]
```diff
@@ -368,3 +368,64 @@ CREATE TABLE LANGUAGE_CODES
...
+      RISP_RECALL_INTERVAL DECIMAL(2) NOT NULL,
+      TRANSACTION_ID INTEGER NOT NULL,
+      TRANSACTION_APP_DATE_TIME DATETIME NOT NULL,
+      TRANSACTION_USER_ORG_ROLE_ID INTEGER NOT NULL,
+      TRANSACTION_DB_DATE_TIME DATE NOT NULL,
+      IGNORE_SELF_REFERRALS BIT NOT NULL,
+      IGNORE_GP_REFERRALS BIT NOT NULL,
+      IGNORE_EARLY_RECALL BIT NOT NULL,
+      IS_ACTIVE BIT NOT NULL,
+      LOWER_AGE_RANGE DECIMAL(2) NOT NULL,
```
Comment (Warren-Pitterson) [inline application/CohortManager/Set-up/database/create_statement.sql:397]: There are seveeral properties here which i think are wrong (not your fault) but i don't see why an upper age would be a decimal? should be a whole number, probably TINYINT is more suitable 

TINYINT: If the age range is between 0 and 255, which is suitable for most applications.
SMALLINT: If you need a larger range, from -32,768 to 32,767.
INT: If you need an even larger range, from -2,147,483,648 to 2,147,483,647, though this is usually overkill for age.

File: application/CohortManager/Set-up/database/create_statement.sql (hunk 3 above)
Comment (CIJones11) [reply application/CohortManager/Set-up/database/create_statement.sql]: Changed to VARCHAR, thanks mate

File: application/CohortManager/Set-up/database/create_statement.sql [hunk 5]
```diff
@@ -368,3 +368,64 @@ CREATE TABLE LANGUAGE_CODES
...
+)
+BEGIN
+    CREATE TABLE [dbo].[BSO_ORGANISATIONS]
+    (
+      BSO_ORGANISATION_ID INT IDENTITY(1,1) PRIMARY KEY,
+      BSO_ORGANISATION_CODE VARCHAR(4) NOT NULL,
+      BSO_ORGANISATION_NAME VARCHAR(60) NOT NULL,
+      SAFETY_PERIOD DECIMAL(2) NOT NULL,
+      RISP_RECALL_INTERVAL DECIMAL(2) NOT NULL,
+      TRANSACTION_ID INTEGER NOT NULL,
```
[the remark this answers is not in the document]
Comment (CIJones11) [reply application/CohortManager/Set-up/database/create_statement.sql]: Using INT now

File: application/CohortManager/Set-up/database/create_statement.sql (hunk 4 above)
Comment (CIJones11) [reply application/CohortManager/Set-up/database/create_statement.sql]: Changed it so it's matching what you mention there, thanks!

**Review by Warren-Pitterson — APPROVED** [review]

**Review by SamRobinson75684 — APPROVED** [review]

**Pull request merged**